component/src/main/java/org/wso2/extension/siddhi/io/feed/sink/FeedSink.java

```diff
@@ -0,0 +1,268 @@
+                    optionHolder.validateAndGetStaticValue(Constants.TIME_OUT,
+                            Constants.DEFAULT_TIME_OUT));
+        }
+    }
+
+    @Override
+    public void publish(Object payload, DynamicOptions dynamicOptions) throws ConnectionUnavailableException {
+        HashMap<String, String> map = (HashMap) payload;
+        ClientResponse resp = null;
+        switch (atomFunc) {
+            case Constants.FEED_CREATE: {
+                Entry entry = EntryUtils.createEntry(map, abdera.newEntry());
+                entry.setPublished(new Date());
+                try {
+                    resp = abderaClient.post(url.toString(), entry);
+                } catch (RuntimeException exception) {
+                    throw new FeedErrorResponseException("Connection timeout exception in " + streamDefinition.getId()
+                            + ". The host did not accept the connection within timeout of "
+                            + abderaClient.getConnectionTimeout());
+                }
+                break;
+            }
+            case Constants.FEED_DELETE: {
+                try {
+                    resp = abderaClient.delete(map.get("id"));
+                } catch (RuntimeException exception) {
+                    throw new FeedErrorResponseException("Connection timeout exception in " + streamDefinition.getId()
+                            + ". The host did not accept the connection within timeout of "
+                            + abderaClient.getConnectionTimeout());
+                }
+                break;
+            }
+            case Constants.FEED_UPDATE: {
+                try {
+                    resp = abderaClient.get(url.toString());
+                    Document<Entry> doc = resp.getDocument();
+                    Entry entry = doc.getRoot();
+                    entry = EntryUtils.createEntry(map, entry);
+                    resp = abderaClient.put(url.toString(), entry);
+                }  catch (RuntimeException exception) {
+                    throw new FeedErrorResponseException("Connection timeout exception in " + streamDefinition.getId()
+                            + ". The host did not accept the connection within timeout of "
+                            + abderaClient.getConnectionTimeout());
+                }
+                break;
+            }
+        }
+
+        if (resp != null) {
+            if (resp.getStatus() != httpResponse) {
+                throw new FeedErrorResponseException("Response status conflicts in " + streamDefinition.getId()
+                        + " response status code is : " + resp.getStatus() + "-" + resp.getStatusText());
+            }
+            resp.release();
+        } else {
+            throw new FeedErrorResponseException("Response is null");
+        }
+    }
+
+    @Override
+    public void connect() throws ConnectionUnavailableException {
+
+    }
+
+    @Override
+    public void disconnect() {
+
```

Comment:
Once you update the extension to create the connection in the connect method, use this method to disconnect it and clean